### DISCUSSION TITLE
TWAI v2 missing for ESP32 C6

### DISCUSSION THREAD

**Discussion by rand12345** [espressif/arduino-esp32#10107] [Q&A] [answered]
I have been working with the ESP32 C6 module and have encountered a lack of implementation for multiple TWAI drivers. According to the [ESP-IDF documentation](https://docs.espressif.com/projects/esp-idf/en/stable/esp32c6/api-reference/peripherals/twai.html#install-multiple-twai-instances), it is possible to install multiple TWAI instances to use both TWAI0 and TWAI1 on the ESP32 C6. However, it seems that the current Arduino-ESP32 library does not yet support this v2 feature.

Attempting to [configure separate TWAI interfaces](https://docs.espressif.com/projects/esp-idf/en/stable/esp32c6/api-reference/peripherals/twai.html#install-multiple-twai-instances) fails with:

```

error: 'twai_handle_t' was not declared in this scope; did you mean 'twai_mode_t'?
  183 |   twai_handle_t twai_bus;
      |   ^~~~~~~~~~~~~
      |   twai_mode_t

```

Is there a timeline or planned release date for when the multiple TWAI drivers will be fully implemented for the ESP32 C6 module?

**Comment by P-R-O-C-H-Y** [comment, marked as answer]
Hi @rand12345, this feature is available from ESP-IDF 5.2 version ([docs](https://docs.espressif.com/projects/esp-idf/en/release-v5.1/esp32c6/api-reference/peripherals/twai.html#two-wire-automotive-interface-twai) for v5.1, see the warning note). The current Arduino core (3.0.X) is based on ESP-IDF v5.1. The new minor version of Arduino core (3.1.X) should be based on ESP-IDF v5.3, so in that release the feature of multiple TWAI drivers will be available.
